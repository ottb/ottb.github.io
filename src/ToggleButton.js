import React, { Component } from 'react';

class ToggleButton extends Component {
  constructor(props) {
    super(props);
    this.state = {isToggled: false};
    this.togglePosition = this.togglePosition.bind(this);
  }

  togglePosition() {
    this.setState({isToggled: !this.state.isToggled});
    this.props.toggleDisplay();
  }
  
  render() {
    var buttonStyle = {
        position: "fixed",
        width: "100%",
        height: "60px",
        backgroundColor: "rgba(0,117,180, 1)",
        padding: "5px",
        bottom: "0px"
    }
    return (
        <div>
            <button style={buttonStyle} onClick={this.togglePosition}>{this.props.text} ({String(this.state.isToggled)})</button>
        </div>
    );
  }
}

export default ToggleButton;