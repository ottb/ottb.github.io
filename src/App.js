import React, { Component } from 'react';
import HTML5Video from './HTML5Video';
import KalturaVideo from './KalturaVideo';
import ToggleButton from './ToggleButton';
import PageContent from './PageContent';
import './styles.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {showContent: false};
    this.toggleDisplay = this.toggleDisplay.bind(this);
  }

  // this probably isn't the best place for this; likely need something between app and components
  toggleDisplay() {
    this.setState({
      showContent: !this.state.showContent
    });
  }

  render() {
    var userAgent = navigator.userAgent || navigator.vendor || window.opera;
    var isMobile = ('ontouchstart' in document.documentElement && /Mobi/.test(userAgent));
    return(
      <div>
        {!this.state.showContent && (isMobile ? <KalturaVideo /> : <HTML5Video />)}
        <ToggleButton text="2016 Annual Report" toggleDisplay={this.toggleDisplay.bind(this)} />
        {this.state.showContent && <PageContent />}
      </div>
    );
  }
}

export default App;
