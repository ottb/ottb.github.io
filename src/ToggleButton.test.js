import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ToggleButton from './ToggleButton';
import App from './App';

function toggleDisplay() {

}

it('button changes isToggled state when clicked', () => {
  const component = renderer.create(
    <ToggleButton text="2016 Annual Report" />
  );
  let tree = component.toJSON();
  expect(tree.children[0]).toMatchSnapshot();

  //manually trigger click
  tree.children[0].props.onClick();
  //re-rendering
  tree = component.toJSON();
  expect(tree.children[0]).toMatchSnapshot();

  //manually trigger click
  tree.children[0].props.onClick();
  //re-rendering
  tree = component.toJSON();
  expect(tree.children[0]).toMatchSnapshot();
});
