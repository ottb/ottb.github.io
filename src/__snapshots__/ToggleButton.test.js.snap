exports[`test button changes isToggled state when clicked 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(0,117,180, 1)",
      "bottom": "0px",
      "height": "60px",
      "padding": "5px",
      "position": "fixed",
      "width": "100%",
    }
  }>
  2016 Annual Report
   (
  false
  )
</button>
`;
